File: Week-2/SalesByMatch/src/main/java/MyCode.java
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class MyCode {
    public static int sockMerchant(int n, List<Integer> ar) {
        Set<Integer> arrSet = ar.stream().collect(Collectors.toSet());
        int counter = 0;
        for (int element: arrSet){
            counter += ar.stream().filter( i -> i == element).collect(Collectors.toList()).size()/2;
        }
        return counter;
    }
}

File: Week-2/SalesByMatch/src/main/kotlin/MyCode.kt
fun sockMerchant(n: Int, ar: Array<Int>): Int {
    val arToSet = ar.toSet()
    var counter = 0
    arToSet.forEach{
            i -> counter += ar.filter{it==i}.size / 2
    }


    return counter

}

fun main(args: Array<String>) {
    val n = readLine()!!.trim().toInt()

    val ar = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    val result = sockMerchant(n, ar)

    println(result)
}